### PR TITLE
lib/discover: Add instance ID to local discovery (fixes #3278)

### LINKED ISSUE #3278: Improve local discovery reaction time at startup
Once we refactor the local discovery protocol we should provide a reliable mechanism to detect a device that is restarting, and provide it with up to date local discovery information.

One idea on how to do this is to include a sequence number in our announcement packets. When a device receives an announcement from another device that it hasn't seen before, or an announcement with a lower sequence number that what it has previously seen for that device, it will respond with an announcement packet of it's own.

This resolves an issue where devices that restart would otherwise only have global discovery data for the first few moments and thus prefer external connections over local ones.

### PR DESCRIPTION
**Pull request by calmh**
A random "instance ID" is generated on each start of the local discovery
service. The instance ID is included in the announcement. When we see a
new instance ID we treat is a new device and respond with an
announcement of our own. Hence devices get to know each other quickly on
restart.

There's a unit test for the "considered new device" logic.

### CODE REVIEW

File: lib/discover/discover.go [hunk 1]
```diff
@@ -26,6 +26,7 @@ type CacheEntry struct {
 	when       time.Time // When did we get the result
 	found      bool      // Is it a success (cacheTime applies) or a failure (negCacheTime applies)?
 	validUntil time.Time // Validity time, overrides normal calculation
+	instanceID int64     // for local discovery, the instance ID (random on each restart)
```
Comment (xduugu) [inline lib/discover/discover.go]: Does this need to be first in the struct as well or is the alignment correct for 32-bit builds?

**Comment by AudriusButkevicius** [comment]
@st-review merge

**Comment by st-review** [comment]
:ok_hand: Merged as 9a0e5a7c183623e74905e1fd5b5abab4bdcfd150. Thanks, @calmh!

**Comment by AudriusButkevicius** [comment]
Needs docs

**Pull request closed without merging**